src/fal/packages/bridge.py

```diff
@@ -32,6 +32,9 @@ def accept(self) -> ConnectionWrapper:
 
 
 def controller_connection() -> _DillListener:
+    # The controller here assumes that there will be at most one
+    # client. This restriction might change in the future as an
+    # optimization.
```

Comment:
Should we do a singleton? I mean to actually make sure it there is only one.

Reply:
This is not a singleton though, there might be multiple controller connections but each one of them will only wait for one child to connect (N controllers, N children. Instead of 1 controller and N children in the future if we ever come to that point).